### ISSUE TITLE
Flake test : TestDeploymentController_dontSyncDeploymentsWithEmptyPodSelector

### ISSUE THREAD

**Issue by k82cn** [kubernetes/kubernetes#34601]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
Bug report.

**Kubernetes version** (use `kubectl version`):
master

**Environment**:
- **Others**: CI envionment

**What happened**:

```
=== RUN   TestDeploymentController_dontSyncDeploymentsWithEmptyPodSelector
E1012 02:19:51.546404   12596 reflector.go:311] pkg/controller/informers/factory.go:73: Failed to watch *extensions.ReplicaSet: unhandled watch: core.WatchActionImpl{ActionImpl:core.ActionImpl{Namespace:"", Verb:"watch", Resource:unversioned.GroupVersionResource{Group:"extensions", Version:"", Resource:"replicasets"}, Subresource:""}, WatchRestrictions:core.WatchRestrictions{Labels:labels.internalSelector{}, Fields:fields.andTerm{}, ResourceVersion:""}}
E1012 02:19:51.556930   12596 reflector.go:311] pkg/controller/informers/factory.go:73: Failed to watch *api.Pod: unhandled watch: core.WatchActionImpl{ActionImpl:core.ActionImpl{Namespace:"", Verb:"watch", Resource:unversioned.GroupVersionResource{Group:"", Version:"", Resource:"pods"}, Subresource:""}, WatchRestrictions:core.WatchRestrictions{Labels:labels.internalSelector{}, Fields:fields.andTerm{}, ResourceVersion:""}}
E1012 02:19:51.561232   12596 reflector.go:311] pkg/controller/informers/factory.go:73: Failed to watch *extensions.Deployment: unhandled watch: core.WatchActionImpl{ActionImpl:core.ActionImpl{Namespace:"", Verb:"watch", Resource:unversioned.GroupVersionResource{Group:"extensions", Version:"", Resource:"deployments"}, Subresource:""}, WatchRestrictions:core.WatchRestrictions{Labels:labels.internalSelector{}, Fields:fields.andTerm{}, ResourceVersion:""}}
--- FAIL: TestDeploymentController_dontSyncDeploymentsWithEmptyPodSelector (0.03s)
	deployment_controller_test.go:270: expected deployment controller to not take action
```

**How to reproduce it** (as minimally and precisely as possible):
Flake test on https://github.com/kubernetes/kubernetes/pull/33546 (https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/33546/kubernetes-pull-test-unit-integration/48867/), but can not reproduce locally (Mac).

**Comment by lukaszo** [comment]
I think it's a duplicate of already fixed #34547

**Comment by piosz** [comment]
Closing in favor of #34580
